plaso/engine/artifact_filters.py

```diff
@@ -65,47 +74,38 @@ def CheckKeyCompatibility(key_path):
         'Prefix of key "{0:s}" is currently not supported'.format(key_path))
     return False
 
-  def BuildFindSpecs(self, environment_variables=None):
+  def BuildFindSpecs(self, artifact_filter_names, environment_variables=None):
```

Comment:
We don't have to change it now, but this method isn't needed any more, and the logic can just move to the initializer.

Reply:
Let's chat about this, it not entirely clear to me what you mean.

Reply:
We can just pass artifact_filter_names and environment in __init__ and we don't need this method. In the engine, which is the only place ArtifactDefinitionsFilterHelper is initialized outside of tests, we call the methods directly after each other. 
Filed https://github.com/log2timeline/plaso/issues/2438 for this (and other changes).